CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+#[[
+This source file is part of the Swift System open source project
+
+Copyright (c) 2019 Apple Inc. and the Swift Numerics project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+#]]
+
+cmake_minimum_required(VERSION 3.16.0)
+project(swift-system
+  LANGUAGES C Swift)
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
+
+include(SwiftSupport)
+
+add_subdirectory(Sources)
+
+get_property(SWIFT_SYSTEM_EXPORTS GLOBAL PROPERTY SWIFT_SYSTEM_EXPORTS)
+export(TARGETS ${SWIFT_SYSTEM_EXPORTS}
+  NAMESPACE SwiftSystem::
+  FILE swift-system-config.cmake
+  EXPORT_LINK_INTERFACE_LIBRARIES)
```

Comment:
What does this `export` do?

Reply:
This export is the trick that enables wiring up the entire toolchain build together.  It creates a `swift-system-config.cmake` which can then be used for any subsequent build to just grab the build artifacts from the build tree without having to install and create a single distribution image and incrementally add content into it.  Once everything is built and verified to work together, you just create a single image that you intend to distribute.